### PR TITLE
refactor: game bridge to use metrics trackError and trackDuration methods

### PR DESCRIPTION
**Pull request by CodeSchwert**
### Hi👋, please prefix this PR's title with:

- [ ] `breaking-change:` if you have introduced modification that necessitates immediate adjustments by this SDK's users to their applications, clients, or integrations to avert disruptions to existing features or functionalities.
- [x] `feat:`, `fix:`, `refactor:`, `docs:`, or `chore:`.

# Summary

[DX-2970](https://immutable.atlassian.net/browse/DX-2970)
[DX-2971](https://immutable.atlassian.net/browse/DX-2971)

Switch to the new metrics `trackError` and `trackDuration` methods.

Game bridge functions have also been refactored to throw errors instead of setting a string error message in the callback function, so all errors are now handled in the `catch` block.

# Detail and impact of the change
- Error stack should give better output as to which function call raised the error.

## Changed
- Game bridge function errors now handled consistently

[DX-2970]: https://immutable.atlassian.net/browse/DX-2970?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DX-2971]: https://immutable.atlassian.net/browse/DX-2971?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/game-bridge/src/index.ts [hunk 1]
```diff
@@ -78,13 +83,29 @@ declare function blu_event(event: string, data: string): void;
...
 declare function UnityPostMessage(message: string): void;
 
-const callbackToGame = (data: object) => {
+interface CallbackToGameData {
+  responseFor: string;
+  requestId: string;
+  success: boolean;
+  error?: string;
+  errorType?: string | null;
+  code?: string;
```
Comment (nattb8) [inline packages/game-bridge/src/index.ts:92]: `code`, `deviceCode`, `url` and `interval` are only returned for initDeviceFlow. Could they not be part of this CallbackToGameData? Same for `accounts`

Comment (CodeSchwert) [reply packages/game-bridge/src/index.ts]: Added a union type instead so it's clearer

**Review by YermekG — APPROVED** [review]

File: packages/game-bridge/src/index.ts [hunk 2]
```diff
@@ -78,13 +83,36 @@ declare function blu_event(event: string, data: string): void;
...
-const callbackToGame = (data: object) => {
+// Type for common callback data
+type CommonCallbackData = {
+  responseFor: string;
+  requestId: string;
+  success: boolean;
+  error: string | null;
+  errorType?: string | null;
+  result?: any;
+  accounts?: string[];
```
Comment (nattb8) [inline packages/game-bridge/src/index.ts:94]: accounts is not comment too, only used by request accounts. Should it be another type?

Comment (CodeSchwert) [reply packages/game-bridge/src/index.ts]: Added another type for request accounts

**Review by nattb8 — APPROVED** [review]

**Pull request merged**